### ISSUE TITLE
Regression involved 2D parameter array bindings

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#12329]
@perost, please check the latest [master regression report](https://libraries.openmodelica.org/branches/history/master/2024-04-26%2011:30:21..2024-04-27%2003:22:22.html). There are several models failing with [errors](https://libraries.openmodelica.org/branches/history/master/2024-04-26%2011:30:21..2024-04-27%2003:22:22.html) like:
```
Error: Type mismatch in binding 'Temp_nom = fill(288.15, 3)', expected array dimensions [3, 3], got [3].
```
Also the [DeltaRobot](https://libraries.openmodelica.org/branches/master/DeltaRobot/DeltaRobot.html) tests got badly broken. They fail at compile time with [errors](https://libraries.openmodelica.org/branches/master/DeltaRobot/files/DeltaRobot_DeltaRobot.Scenarios.DeltaRobot_IdealActPos_PositionValidation.err) such as:
```
In file included from DeltaRobot_DeltaRobot.Scenarios.DeltaRobot_IdealActPos_PositionValidation.c:7:
In file included from ./DeltaRobot_DeltaRobot.Scenarios.DeltaRobot_IdealActPos_PositionValidation_model.h:28:
./DeltaRobot_DeltaRobot.Scenarios.DeltaRobot_IdealActPos_PositionValidation_functions.h:81:1: error: unknown type name 'DeltaRobot_Types_DiscParameters'
DeltaRobot_Types_DiscParameters omc_DeltaRobot_Types_DiscParameters (threadData_t *threadData, modelica_real omc_D, modelica_real omc_De, modelica_real omc_thickness, modelica_real omc_mass, modelica_real omc_beta);
```
but I guess the root cause is in the frontend, as nothing has changed in the backend/codegen since the previous regression test.

Can you please check?

**Comment by perost** [comment]
It's probably caused by start attributes from types that we don't handle correctly, i.e.:
```
model M
  type T = Real(start = 1.0);
  parameter T x[3];
end M;
```

**Comment by casella** [comment]
Looking at the latest [regression report](http://libraries.openmodelica.org/branches/history/master/2024-04-30%2012:14:30..2024-04-30%2022:30:12.html), the DroneLibrary regressions were fixed by #12354, but not the DeltaRobot ones.

**Comment by casella** [comment]
Thanks @perost, [DeltaRobot](https://libraries.openmodelica.org/branches/master/DeltaRobot/DeltaRobot.html) [back on track](https://libraries.openmodelica.org/branches/history/master/2024-04-30%2022:30:12..2024-05-02%2022:29:34.html)
